File: poc/serasa-rv/src/count-nums.py
from pyspark import SparkConf, SparkContext
conf = (SparkConf()
         .setMaster("local[4]")
         .setAppName("MyApp")
         .set("spark.executor.memory", "1g")
         .set('spark.local.dir', './target/tmp'))
sc = SparkContext(conf = conf)

words = sc.textFile("./gen/data/nums")
#print 'count', words.count()

root = words.map(lambda line: (int(line), 1))

root.reduceByKey(lambda a, b: a + b) \
  .saveAsTextFile('./target/result')

root.reduceByKey(lambda a, b: a * b) \
  .saveAsTextFile('./target/result2')

sc.stop()

File: poc/spark/java/src/main/java/com/murerz/sandbox/spark/CountNums.java
package com.murerz.sandbox.spark;

import java.io.File;

import org.apache.commons.io.FileUtils;
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

import scala.Tuple2;

public class CountNums {

	public static void main(String[] args) throws Exception {
		SparkConf conf = new SparkConf().setAppName("MyApp").setMaster("local[4]");
		JavaSparkContext sc = new JavaSparkContext(conf);

		for (int i = 0; i < 10; i++) {
			System.out.println("Try: " + i);
			FileUtils.deleteDirectory(new File("./target/result"));
			FileUtils.deleteDirectory(new File("./target/result2"));
			FileUtils.deleteDirectory(new File("./target/result3"));

			JavaRDD<String> words = sc.textFile("../gen/data/nums");

			JavaPairRDD<Integer, Integer> root = words.mapToPair(line -> new Tuple2<>(Integer.parseInt(line), 1));
			root.saveAsTextFile("./target/result3");

			root.reduceByKey((a, b) -> a + b).saveAsTextFile("./target/result");
			root.reduceByKey((a, b) -> a * b).saveAsTextFile("./target/result2");
		}

		sc.stop();
	}

}
